File: error-401-423.js
module.exports = {
  handleErrors(tableau, nombre) {
    if (parseInt(nombre) < 1) {
      throw new Error("Erreur 401 : Non autorisé (nombre plus petit que 1)");
    } else if (parseInt(nombre) === 0) {
      throw new Error("Erreur 404 : Fichier non trouvé (nombre égal à 0)");
    } else if (isNaN(parseInt(nombre))) {
      // Utiliser setTimeout pour gérer la demande expirée après 5 secondes
      let nombreEntree = null;
      try {
        nombreEntree = prompt.timeout(5000);
      } catch (error) {
        throw new Error("Erreur 408 : Request Timeout (l'utilisateur n'a pas entré un nombre pendant 5 secondes)");
      }
    } else if (parseInt(nombre) > 10) {
      throw new Error("Erreur 423 : Fichier trop volumineux (nombre plus grand que 10)");
    }
  },
};

File: error_401-423.py
import time

def handle_errors(tableau, nombre):
    int_value = int(nombre)

    if int_value < 1:
        raise ValueError("Erreur 401 : Non autorisé (nombre plus petit que 1)")
    elif int_value == 0:
        raise ValueError("Erreur 404 : Fichier non trouvé (nombre égal à 0)")
    elif isnan(int_value):
        # Utiliser time.sleep pour gérer la demande expirée après 5 secondes
        nombre_entree = None
        try:
            time.sleep(5)
            nombre_entree = int(input("Entrez un nombre : "))
        except ValueError:
            raise ValueError("Erreur 408 : Request Timeout (l'utilisateur n'a pas entré un nombre pendant 5 secondes)")
    elif int_value > 10:
        raise ValueError("Erreur 423 : Fichier trop volumineux (nombre plus grand que 10)")

# La fonction isnan pour vérifier si le nombre est NaN
def isnan(value):
    try:
        import math
        return math.isnan(value)
    except:
        return False
